### PR TITLE
Improve API for cluster bootstrap, remove schema cache, add logger and a little more

### PR DESCRIPTION
**Pull request by antonkharenko**

### CODE REVIEW

File: cluster/src/test/java/io/servicefabric/cluster/examples/ClusterBootstrapExamples.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package io.servicefabric.cluster.examples;
```
Comment (ronenhamias) [inline cluster/src/test/java/io/servicefabric/cluster/examples/ClusterBootstrapExamples.java]: IMO we should move examples to separated module

Comment (antonkharenko) [inline cluster/src/test/java/io/servicefabric/cluster/examples/ClusterBootstrapExamples.java]: Yes-yes, it is going to be separate PR - I just put it somewhere for now

**Comment by ronenhamias** [comment]
+1

**Comment by myroslavlisniak** [comment]
I prefer to merge only squashed commits

**Comment by antonkharenko** [comment]
Agree, but in this PR I have done several different things, some commits may be squashed, but most of them are separate.
